animations-dbus/animations-dbus-server-surface.c

```diff
@@ -0,0 +1,691 @@
+                                                                             GError                      **error)
+{
+  /* Other effects take priority over this one, since server effects
+   * can be overridden by the user. */
+  if (!animations_dbus_server_surface_attach_effect_with_queue_func (server_surface,
+                                                                     event,
+                                                                     server_animation_effect,
+                                                                     g_queue_push_tail,
+                                                                     error))
+    return FALSE;
+
+  return TRUE;
+}
+
+/**
+ * animations_dbus_server_surface_highest_priority_attached_effect_for_event:
+ * @server_surface: The #AnimationsDbusServerSurface with the attached effects.
+ * @event: The event to get the highest priority effect on.
+ *
+ * Check the attached effects for @event on @surface and get the highest priority
+ * effect, returning its #AnimationsDbusServerSurfaceAttachedEffect (private data
+ * for the effect.
+ *
+ * Returns: (transfer none): The #AnimationsDbusServerSurfaceAttachedEffect for the highest
+ *          priority effect on @surface for @event, or %NULL if no events are attached to
+ *          @effect.
+ */
+AnimationsDbusServerSurfaceAttachedEffect *
+animations_dbus_server_surface_highest_priority_attached_effect_for_event (AnimationsDbusServerSurface *server_surface,
+                                                                           const char                  *event)
+{
+  AnimationsDbusServerSurfacePrivate *priv =
+    animations_dbus_server_surface_get_instance_private (server_surface);
+  GQueue *attached_effects_for_event = g_hash_table_lookup (priv->attached_effects_for_events, event);
+
+  if (attached_effects_for_event)
+    {
+      AttachedEffectInfo *info = g_queue_peek_head (attached_effects_for_event);
+
+      if (info == NULL)
+        return NULL;
+
+      return info->attached_effect;
+    }
+
+  return NULL;
+}
+
+/* /com/endlessm/Libanimation/AnimationManager/N/AnimationEffect/M */
+#define EFFECT_PATH_EXPECTED_COMPONENTS 8
+#define EFFECT_PATH_ANIMATION_MANAGER_INDEX 5
+#define EFFECT_PATH_ANIMATION_EFFECT_INDEX 7
+
+static gboolean
+parse_effect_path (const char    *effect_path,
+                   unsigned int  *out_animation_manager_id,
+                   unsigned int  *out_animation_effect_id,
+                   GError       **error)
+{
+  g_auto(GStrv) effect_path_components = g_strsplit (effect_path, "/", 0);
+  unsigned int path_length = g_strv_length (effect_path_components);
+  guint64 animation_manager_id64 = 0;
+  guint64 animation_effect_id64 = 0;
+  g_autoptr(GError) local_error = NULL;
+
+  g_return_val_if_fail (out_animation_manager_id != NULL, FALSE);
+  g_return_val_if_fail (out_animation_effect_id != NULL, FALSE);
```

Comment:
If you declare these at the top, then you save it from doing any work in splitting the string and counting the tokens...

Reply:
True, I'll move these